### PR TITLE
Deprecate list support in Parser

### PR DESCRIPTION
**Pull request by shunyao643**
Remove support for parsing <code>list t/\`TYPES\`</code> in Parser. 

Added additional test cases for Parser#splitCommandTerm

Update DeveloperGuide.md with additional documentation on Design and Architecture

### CODE REVIEW

File: src/test/java/seedu/parser/ParserTest.java [hunk 1]
```diff
@@ -36,8 +36,26 @@ void splitCommandTerm_validCommand_success() throws IncompleteCommandException {
...
+        ArrayList<String> expectedResult = new ArrayList<>(
+                Arrays.asList("list", null)
+        );
+        ArrayList<String> actualResult = parser.splitCommandTerm(
+                "list");
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void splitCommandTerm_listCommandExtraWordsRetained_success() throws IncompleteCommandException {
```
Comment (Bryan-BC) [inline src/test/java/seedu/parser/ParserTest.java:52]: Is this test for the list t/`type`? if it is, do we still need it?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -77,15 +82,15 @@ For added safeguards in equipment deletion, the delete command implements a more
...
 
-Throughout the `Parser` implementation, exceptions were used to return `IncorrectCommand` classes that can be used to pass error messages to the user. These will be discussed in the following segments.
+Throughout the `Parser` implementation, exceptions were caught to return `IncorrectCommand` classes that can be used to pass error messages to the user. These will be discussed in the following segments.
 
 ### Command classes
 
 #### Current Design and Implementation
 
-`Command` classes are largely similar to each other. The most complex command, `UpdateCommand` will be explained here.
+`Command` classes are largely similar to each other. Taking in input Strings from `Parser`, they invoke methods from `EquipmentManager` to perform the necessary actions. Where applicable, the `Command` classes also convert values to the necessary data types (Double, EquipmentType and LocalDate) prior to passing these values in.
```
Comment (yuhaochua) [inline docs/DeveloperGuide.md:91]: Thanks for adding on to the explanation for this portion!

**Review by yuhaochua — APPROVED** [review]
LGTM, good work on updating the developer guide!

File: src/test/java/seedu/parser/ParserTest.java (hunk 1 above)
Comment (shunyao643) [reply src/test/java/seedu/parser/ParserTest.java:52]: It was originally for that, but I've modified and retained it here just as an additional test. It's expected that a command <code>list t/\`SPEAKER\`</code> will return the ArrayList of <code>"list"</code> and <code>t/\`SPEAKER\`</code>. This is to be processed by `parseCommand`.

**Pull request merged**